src/main/java/uk/nhs/prm/repo/ehrtransferservice/repo_incoming/RepoIncomingService.java

```diff
@@ -1,78 +1,78 @@
         final Optional<UUID> nemsMessageId = Optional.of(
             UUID.fromString(repoIncomingEvent.getNemsMessageId())
         );
 
+        conversationActivityService.captureConversationActivityTimestamp(inboundConversationId);
+
         transferService.createConversation(repoIncomingEvent);
 
         gp2gpMessengerService.sendEhrRequest(repoIncomingEvent);
         transferService.updateConversationTransferStatus(inboundConversationId, INBOUND_REQUEST_SENT);
 
         auditService.publishAuditMessage(inboundConversationId, INBOUND_REQUEST_SENT, nemsMessageId);
-        waitForTransferTrackerDbToUpdate(inboundConversationId);
+        waitForConversationToComplete(inboundConversationId);
     }
 
-    private void waitForTransferTrackerDbToUpdate(UUID inboundConversationId)
-        throws InterruptedException, EhrResponseFailedException, EhrResponseTimedOutException {
-        int pollCount = 0;
-        String transferState;
-
-        log.info("Awaiting for transfer status to become INBOUND_COMPLETE for Inbound Conversation ID {}",
-            inboundConversationId.toString().toUpperCase());
+    private void waitForConversationToComplete(UUID inboundConversationId) throws InterruptedException {
+        log.info("Awaiting Inbound Conversation ID {} to complete", inboundConversationId.toString().toUpperCase());
 
-        do {
-            Thread.sleep(ehrResponsePollPeriodMilliseconds);
-            transferState = transferService.getConversationTransferStatus(inboundConversationId);
-
-            log.info("Transfer status yet to be updated to INBOUND_COMPLETE (currently {}) for Inbound Conversation ID {} - attempt {} of {}",
-                transferState,
-                inboundConversationId.toString().toUpperCase(),
-                pollCount + 1,
-                ehrResponsePollLimit
-            );
-
-            if (transferState.equals(INBOUND_COMPLETE.name())) {
-                log.info("Transfer status has been set to INBOUND_COMPLETE for Inbound Conversation ID {}", inboundConversationId.toString().toUpperCase());
-                return;
-            }
-
-            pollCount++;
-        } while (pollCount < ehrResponsePollLimit);
+        while (conversationActivityService.isConversationActive(inboundConversationId)) {
+            Thread.sleep(pollPeriodMilliseconds);
+            verifyConversationNotTimedOut(inboundConversationId);
+        }
+    }
 
-        log.warn("The transfer for Inbound Conversation ID {} has timed out", inboundConversationId.toString().toUpperCase());
-        transferService.updateConversationTransferStatus(inboundConversationId, INBOUND_TIMEOUT);
-        throw new EhrResponseTimedOutException(transferState);
+    private void verifyConversationNotTimedOut(UUID inboundConversationId) {
+        if (conversationActivityService.isConversationTimedOut(inboundConversationId, inboundTimeoutSeconds)) {
```

Comment:
Should `inboundTimeoutSeconds` be passed in, if it's only ever used within this method?